modules/prompt/functions/prompt_sorin_setup

```diff
@@ -71,30 +79,46 @@ function prompt_sorin_precmd {
   # Define prompts.
   RPROMPT='${editor_info[overwrite]}%(?:: %F{1}⏎%f)${VIM:+" %B%F{6}V%f%b"}'
 
-  # Kill the old process of slow commands if it is still running.
-  if (( _prompt_sorin_precmd_async_pid > 0 )); then
-    kill -KILL "$_prompt_sorin_precmd_async_pid" &>/dev/null
-  fi
+  # Kill any currently running async jobs
+  async_flush_jobs 'prompt_sorin'
 
-  # Compute slow commands in the background.
-  trap prompt_sorin_git_info WINCH
-  prompt_sorin_precmd_async &!
-  _prompt_sorin_precmd_async_pid=$!
+  # We need the worker's env to be up to date, so here we are restarting the worker
+  async_stop_worker 'prompt_sorin'
+  async_start_worker 'prompt_sorin' -n -u
+  async_register_callback 'prompt_sorin' _prompt_sorin_callback
+
+  # Kick off async jobs
+  async_job 'prompt_sorin' _prompt_sorin_async_job "$(pwd)"
 }
 
 function prompt_sorin_setup {
   setopt LOCAL_OPTIONS
   unsetopt XTRACE KSH_ARRAYS
   prompt_opts=(cr percent subst)
-  _prompt_sorin_precmd_async_pid=0
-  _prompt_sorin_precmd_async_data="${TMPPREFIX}-prompt_sorin_data"
 
   # Load required functions.
   autoload -Uz add-zsh-hook
+  autoload -Uz async && async
+
+  # Start async worker
+  # async_start_worker 'prompt_sorin' -n -u
```

Comment:
With zsh-async, the worker is spawned off ideally here, when the prompt initializes. This works fine for grabbing up-to-date git info.

This doesn't work with ruby-info python-info, because the environment changes made by rbenv/rvm/virtualenv are made only on the active terminal, not the worker. I guess those environment changes could be sent to the worker job as parameters but that seemed to me to couple too tightly with python-info and ruby-info, and would possibly break as python-info and ruby-info change in the future.

I commented it out when changing the worker to stop/start on every command and forgot to remove it.